src/scheduler/job_info.c

```diff
@@ -4800,6 +4800,7 @@ void create_res_released(status *policy, resource_resv *pjob)
 		pjob->job->resreq_rel = create_resreq_rel_list(policy, pjob);
 	}
 	selectspec = create_select_from_nspec(pjob->job->resreleased);
+	free_selspec(pjob->execselect);
```

Comment:
Just curious, execselect already gets set inside query_jobs, so why do we need to set it again here?

Reply:
Well this is needed to add the job back to calendar. We use execselect (if it is set) to find node solution in check_nodes.

Reply:
@agrawalravi90 We reset it here since it has changed.  When it is set in query_jobs(), it uses the exec_vnode (all the nodes and all the resources).  Here we are setting it to just the resources we released.

Reply:
ok, so we kind of overload execselect so that get_resresv_spec() can return only the released resources and we only look for them when finding resources for the suspended job?

I feel like we could leverage this for degraded reservations as well, set execselect to only the node(s)/resources which went down so that we only look for them when re-confirming the reservation (the server will have to set resources_released on the reservation for the node(s) which went down).

Anyways, thanks for the clarification!